backend/app/routes/admin/putAdmin.js

```diff
@@ -0,0 +1,36 @@
+const to = require('await-to-js').default;
+const Admin = require('../../models/Admin');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const { firstName, lastName, contact, username } = req.body;
+  const userRecord = await Admin.findOne({
+    email: res.locals.decode.email,
+  });
+
+  if (username === userRecord.username) {
```

Comment:
This doesn't make much sense, What if I am user1 with username --> a
you are user2 with username --> b
I(user1) tries to update my profile using username as 'b'.
Do you think that is acceptable? Does your code handle that? How will the frontend know that it is not possible and the root cause is due to invalid request from user

Reply:
user1 can try to update user2 but will not able to update it because he will not be authorized to do this, or can we use auth for authorization

Reply:
How can user1 be allowed to complete the request?

Reply:
here user1 is not trying to update user2's profile but his own profile using username as 'b'